skfp/descriptors/topological.py

```diff
@@ -0,0 +1,210 @@
+        <https://www.sciencedirect.com/science/article/pii/S0096300315014149>
+
+    Examples
+    --------
+    >>> from rdkit import Chem
+    >>> from descriptors.topological import weiner_index
+    >>> mol = Chem.MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> weiner_index(mol)
+    27
+    """
+    if distance_matrix is None:
+        distance_matrix = GetDistanceMatrix(mol)
+    return np.sum(distance_matrix) // 2
+
+
+def average_weiner_index(mol: Mol, distance_matrix: Optional[np.ndarray] = None) -> float:
+    """
+    Average Wiener Index (AW).
+
+    The implementation uses RDKit to calculate the Average Wiener Index for molecular graphs.
+    It is derived by dividing the Wiener index by the total number of atom pairs in the molecule.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecular graph for which the Average Wiener index is to be calculated.
+
+    distance_matrix : np.ndarray, optional
+        Precomputed distance matrix. If not provided, it will be calculated using RDKit.
+
+    References
+    ----------
+    .. [1] Andrey A. Dobrynin, Ivan Gutman,
+        "The average Wiener index of hexagonal chains",
+        Computers & Chemistry, Volume 23, Issue 6, 1999, Pages 571-576, ISSN 0097-8485,
+        <https://www.sciencedirect.com/science/article/pii/S0097848599000352>
+
+    Examples
+    --------
+    >>> from rdkit import Chem
+    >>> from descriptors.topological import average_weiner_index
+    >>> mol = Chem.MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> average_weiner_index(mol)
+    1.8
+"""
+
+    if distance_matrix is None:
+        distance_matrix = GetDistanceMatrix(mol)
+    wiener_idx = weiner_index(mol, distance_matrix)
+    num_atoms = mol.GetNumAtoms()
+    return (2 * wiener_idx) / (num_atoms * (num_atoms - 1))
+
+
+def graph_distance_index(mol: Mol, distance_matrix: Optional[np.ndarray] = None) -> float:
+    """
+    Graph Distance Index (Tigdi).
+
+    The implementation calculates the Graph Distance Index for molecular graphs.
+    The GDI is a topological descriptor defined as the squared sum of all graph distance
+    counts in the molecular graph's distance matrix. The formula for calculating GDI is given as:
+
+    .. math::
+
+        GDI = \sum_{k=1}^{D} \left(k \cdot f_k\right)^2
+
+    where:
+
+    - :math:`D` is the topological diameter of the graph (the largest graph distance).
+    - :math:`f_k` is the total number of distances in the graph equal to :math:`k`.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecular graph for which the Graph Distance Index is to be calculated.
+
+    distance_matrix : np.ndarray, optional
+        Precomputed distance matrix. If not provided, it will be calculated using RDKit.
+
+    References
+    ----------
+    .. [1] PyChem documentation
+        <https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/pychem/manual.pdf>
```

Comment:
Proper reference is `Rouvray, D. H. In Chemical Applications of Topology and Graph Theory; King, R. B., Ed.; Elsevier: Amsterdam, 1983; pp 159-177.`, or https://pubs.acs.org/doi/10.1021/ci9502461